src/util/jira-utils.ts

```diff
@@ -166,6 +166,10 @@ export const removeSubscription = async (
 		throw new RestApiError(404, "RESOURCE_NOT_FOUND", "Can not find subscription");
 	}
 
+	if (subscription.jiraHost !== jiraHost) {
+		throw new RestApiError(500, "JIRAHOST_MISMATCH", "Jirahosts do not match for this subscription");
```

Comment:
Well, it should be 403 or 401 anyway, cause it not the server's fault that client submit a wrong subscription Id. 
But /nit

Reply:
403 and 401 might give the wrong idea here, coz they indicate that the user has no permission or not authenticated. I think 500 is better suited, so leaving it as it is for now.